src/utils/hibp.js

```diff
@@ -61,8 +76,21 @@ async function _throttledFetch (url, reqOptions, tryCount = 1) {
         throw new InternalServerError(`bad response: ${response.status}`)
     }
   } catch (err) {
-    console.error('_throttledFetch', { err })
-    throw new InternalServerError(getMessage('error-hibp-connect'))
+    const {SERVER_URL, HIBP_KANON_API_SUFFIX_FAKE} = process.env;
+    if (SERVER_URL === undefined || HIBP_KANON_API_SUFFIX_FAKE === undefined) {
+      throw new InternalServerError('Environemnt not configured correctly: Missing server_url or hibp_suffix_fake')
+    }
+  
+    const mockHost = String(SERVER_URL) + String(HIBP_KANON_API_SUFFIX_FAKE);
+    const mockUrl = mockHost + url.match(/\/breachedaccount.*/);
+
+    console.error('_throttledFetch - attempting to switch to mock endpoint', { err })
```

Comment:
I'm a little confused by this... does it attempt to hit the real endpoint and then fall back to the mock?

Reply:
Yes! In the catch statement, instead of just throwing an error I am attempting to switch to the mock endpoint.